node/core/pvf/src/pvf.rs

```diff
@@ -23,26 +23,31 @@ use std::{
 	cmp::{Eq, PartialEq},
 	fmt,
 	sync::Arc,
+	time::Duration,
 };
 
-/// A struct that carries code of a parachain validation function, its hash, and a corresponding
-/// set of executor parameters.
+#[cfg(test)]
+use crate::host::tests::TEST_PREPARATION_TIMEOUT;
+
+/// A struct that carries code of a parachain validation function, its hash, a corresponding
+/// set of executor parameters, and a preparation timeout.
 ///
 /// Should be cheap to clone.
 #[derive(Clone, Encode, Decode)]
-pub struct PvfWithExecutorParams {
+pub struct PvfExhaustive {
```

Comment:
Here's a problem. There is already something called "PVF params", those are params passed to the PVF itself during the execution. So we called environment params the "executor params" and now, if we called the whole thing `PvfWithParams` it would be twice dubious. So if you have any better naming ideas for "PVF-code-blob-and-everything-needed-to-prepare-an-artifact", you're welcome, I'm not a big fan of that "Exhaustive" too, I just failed to invent something more meaningful, and I didn't want to fall back to just `Pvf`, because it is not just PVF anymore.

Reply:
Yes, it is an awkward situation. Naming is a hard problem. Maybe... `PvfWithPreparationData`? Haha, not great, but better I think. Or maybe `PvfPreparationData` or even `PreparationData`. If I'm reading code and I see `PvfExhaustive` by itself, it doesn't tell me anything about what the struct does.